penguin/web/routes.py

```
@@ -2083,6 +2089,197 @@ def _get_coordinator(core: PenguinCore):
+    if manager is not None:
+        return manager
+
+    tool_manager = getattr(core, "tool_manager", None)
+    manager = getattr(tool_manager, "skill_manager", None)
+    if manager is not None:
+        return manager
+
+    raise HTTPException(status_code=503, detail="Skills are not initialized")
+
+
+def _normalize_skill_activation_result(result: Any) -> Dict[str, Any]:
```

Ruff (0.15.12)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `result`

(ANN401)